### PR TITLE
[inductor] Handle floordiv and remainder in IndexPropagation

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #102277
* #102000

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx

### CODE REVIEW

File: test/inductor/test_torchinductor.py [hunk 1]
```diff
@@ -6419,26 +6419,66 @@ def fn(x, n):
...
-                msg=f"Found indirect indexing in code:\n{code}",
+            self.assertEqual(flip_opt(x), flip(x))
+
+        def test_index_propagation_floordiv(self):
+            def repeat_interleave(x, n):
+                # e.g. x=[1, 2, 3], n=2 => returns [1, 1, 2, 2, 3, 3]
+                i = torch.arange(x.shape[0] * n, device=x.device)
+                return x[i // n]
+
+            x = torch.randn(8, device="cuda")
```
Comment (lezcano) [inline test/inductor/test_torchinductor.py:6452]: These should work, in theory, also on CPU. I think we should test them there, and if they don't work, xfail them.

**Review by lezcano — APPROVED** [review]
Just one comment, but otherwise this looks great!

**Comment by kit1980** [comment]
This or the first one in the stack broke inductor tests
https://github.com/pytorch/pytorch/actions/runs/5095190248/jobs/9160028124

A land race probably?

__________________________ CpuTests.test_dropout3_cpu __________________________
Traceback (most recent call last):
  File "/var/lib/jenkins/workspace/test/inductor/test_torchinductor.py", line 5272, in test_dropout3
    self.assertEqual(
  File "/opt/conda/envs/py_3.9/lib/python3.9/site-packages/torch/testing/_internal/common_utils.py", line 3070, in assertEqual
    raise error_metas[0].to_error(
AssertionError: Scalars are not equal!

Expected 5 but got 6.
Absolute difference: 1
Relative difference: 0.2

**Pull request closed without merging**